### ISSUE TITLE
CILG Roadmap 

### ISSUE THREAD

**Issue by austinfrey** [austinfrey/p2p-community-infrastructure#4]
# vision
The initial vison for the CILG was to create place where anyone could learn about what is needed to host their own p2p infrastructure. It's less about protocols, apps, and products and more about identifying the best practices for managing/running hardware in the home, in the community or in the cloud.

# strategy
Identify one use case from the p2p world and provide a happy path to easily setup the infrastructure needed to support that use case. This will include scripts, documentation in many forms i.e. traditional docs, blog style guides, video walk-throughs, online meetups etc etc. (Initially this will focus on SSB Pub setup). Once the CILG has worked through this process with this narrow use case, start identifying other use cases and replicate the process. 

# short term
- initial ssb pub scripts w/ accompanying documentation
  - documentation should include explanations for all commands so a user can become familiar with what's happening on their machine
- scripts and docs will exist as a git repo, github or somewhere else.
- pub setup can be a one line command but could also be done line by line "by hand"
- accompanying tutorial style walk through (written, blog style)

# mid term
- package scripts as a debian package
- publish docs as an actual web page, this web page will serve as a hub for future efforts as well
- video pub setup tutorial
- online gathering/meetups for walk-throughs QA

# long term
- Identify additional use cases and replicate the process. 
- Meat Space workshops 
- Node School style lessons
- Collaborate with other groups with like minded visions/values
- could we offer support to communities looking to get going?

**Comment by treasuretron** [comment]
sweet, love it, I might be able to help with a little bit of everything or whatever is needed. Will try and make the call on the 4th

**Comment by Hurricos** [comment]
Hey there! @treasuretron pointed this out to me. What's this about a call on the 4th?

Comment: To help out a community, we need tools that advance people from every state to every other state - "has no hardware" to "has some hardware", "has no Linux knowledge" to "has some Linux knowledge", "manages no communities" to "manages some communities", etc. I think a part of this project and every other project with aligned goals should be to identify the audience (input) and the expected result (output), then to solicit information from the audience about how effective it has been for them.

**Comment by austinfrey** [comment]
@Hurricos this is good feedback thanks! 

there is a call approaching. either this Saturday or the next. the initial gathering date was meant to be a placeholder so still trying to see whether the 4th or 11th will be best, but I'll post to the tomesh group tomorrow with the updated details and a finalized date. hope to see you there no matter which date it will be :)

**Comment by Hurricos** [comment]
@austinfrey great. I found the link on chat.tomesh.net (%2raRSoYFZk6aocJ0EYzHma9Bp4xw9W/pvXtSKyWYlsk=.sha256) and hope to attend. What time is it at on the 11th?

Edit: Found it, and I've imported it into my calendar. Thanks!

**Comment by benhylau** [comment]
Hello! I am reemerging from holidays :D and sorry I didn't get to this before the break as I promised. I was going to write my hopes & dreams, but since I read this I felt it is already very close to what I'd hope for, so I will add my comments here instead.

I agree with @austinfrey about this as a starting point:

>Identify one use case from the p2p world and provide a happy path to easily setup the infrastructure needed to support that use case.

Focus on one thing that can quickly get adopted by an existing audience. I would also bring potential users to the calls, and have it as a support / feedback session. For example, "I want to start a SSB pub in my community of 20 people" and if we don't have existing resources to get this person started, we know that's a friction point. Perhaps this is more of a mid term goal as @austinfrey wrote in "online gathering/meetups for walk-throughs QA"?

@Hurricos to your point:

>identify the audience (input) and the expected result (output)

I think a good way to do that is invite the potential users to the call, to inform the range of needs that we need to cover. Myself, as a user, for example, would like to see how pubs can be installed and operated completely off the Internet in a local mesh network. It's quite a specific need I have, that may not be apparent to the group until I described the use case, and there probably isn't docs for that.

I also want to add the one long term goal of **Being able to run community infrastructure on networks that are independent from Internet, on hardware that are low-cost and reasonable to run in a common household**, which maps to the long term goal [here](https://github.com/tomeshnet/mesh-services#hopes--dreams).

Other than these comments, I think what @austinfrey wrote is a good starting roadmap for the group.

In fact, I just published the documentation for the DWeb Camp mesh network before the holidays, and it more or less followed the format and timeline of @austinfrey's CILG roadmap, but for a different project. The docs are hosted as a static website on gh-pages, and scripts in another directory in same repo as website. [Section 2](https://dweb-camp-2019.github.io/meshnet/2.0-network-design.html) documents the commands to configure things referencing the scripts, while other sections document other aspects of the project in a blog format. Some of us involved in that project are now discussing gatherings based on the published resources, which maps to the mid/long term goals @austinfrey set. Would like to know if this is the sort of documentational layout and granularity you have in mind, and perhaps we can start thinking about what topics are important in the outline.

Also want to quote this from @mycognosist's post a while back `%/sl6mUFBXNVkU0U8pU0+1uyOQHJ+gjS0xdddEa3V7QY=.sha256`, that I felt makes a good list of topics for video or text format documentation:

>Videos I’d hope to see as a prospective pub operator or p2p tinkerer would be ones like:
>
>Setup a Scuttlebutt pub on a cloud-based server
>Running a Scuttlebutt pub on Raspberry Pi
>Configure Scuttlebutt to run over Tor
>Connect two Scuttlebutt peers with Yggdrasil
>Installing and using git-ssb
>How to deploy a web-viewer for Scuttlebutt

I also "federated" the call series to: https://tomesh.net/2020-01-11/community-infrastructure-learning-group/
